lib/actions/reassemble.js

```diff
@@ -2,7 +2,34 @@
 const { messages } = require('elasticio-node');
 const ObjectStorageWrapperExtended = require('./utils-wrapper/ObjectStorageWrapperExtended');
 
-async function processAction(msg) {
+let timeHandle;
+let groupList = [];
+let delay;
+
+async function timer(this_) {
+  for (let i = 0; i < groupList.length; i += 1) {
+    const storage = new ObjectStorageWrapperExtended(this_);
+    // eslint-disable-next-line no-await-in-loop
+    const results = await storage.lookupParsedObjectById(groupList[i]);
+    const incomingData = {};
+    results.messages.forEach((message) => {
+      incomingData[message.messageId] = message.messageData;
+    });
+
+    // eslint-disable-next-line no-await-in-loop
+    await storage.deleteObjectById(groupList[i]);
+
+    // eslint-disable-next-line no-await-in-loop
+    await this_.emit('data', messages.newMessageWithBody({
+      groupSize: results.messages.length,
+      groupId: results.messages[0].groupId,
+      messageData: incomingData,
+    }));
```

Comment:
In order to preserve at least once delivery, it would be better to emit and then delete.